### DISCUSSION TITLE
MeasureElementTool unit

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#158] [Q&A] [answered]
MeasureElementTool  use display unit is  ft ,how do change unit for example m or cm 
MeasureAreaByPointsTool ,MeasureAreaTool,MeasureVolumeTool have this question too.

**Comment by pmconne** [comment]
[IModelApp.quantityFormatter](https://github.com/imodeljs/imodeljs/blob/06ec145d9e5c15cc70e807b067210d144a5552c9/core/frontend/src/IModelApp.ts#L122) has a `useImperialFormats` property you can set to `false` to force metric display. You can change this whenever you want. If you **always** want metric units, you can configure that at startup:
```ts
const quantityFormatter = new QuantityFormatter();
quantityFormatter.useImperialUnits = false;
const opts: IModelAppOptions = { quantityFormatter };
await IModelApp.startup(opts);
```
[QuantityFormatter](https://github.com/imodeljs/imodeljs/blob/master/core/frontend/src/QuantityFormatter.ts) is currently marked as `alpha`, so it's excluded from the API documentation. I'm not sure whether/how you can tell it to e.g. use cm instead of m - @bsteinbk probably could tell you.

**Reply by bsteinbk** [reply]
Currently the only way to change the formatted units is to create your own subclass of QuantityFormatter and define your own set of default formats for QuantityType.Length and QuantityType.Volume. There are plans to address this limitation but the work is not yet scheduled.

**Comment by paprophet** [comment, marked as answer]
" quantityFormatter.useImperialFormats = false;  " -run error 
 error message 

Unhandled Rejection (TypeError): Cannot read property 'startDefaultTool' of undefined
QuantityFormatter.set useImperialFormats [as useImperialFormats]
 /src/QuantityFormatter.ts:713
  710 |   if (this._activeSystemIsImperial === useImperial)
  711 |     return;
  712 | 
> 713 |   IModelApp.toolAdmin.startDefaultTool();
 | ^  714 |   this._activeSystemIsImperial = useImperial;
  715 |   this.loadFormatAndParsingMaps(useImperial); // eslint-disable-line @typescript-eslint/no-floating-promises
  716 | }

**Reply by pmconne** [reply]
Show your code. Did you await `IModelApp.startup()` before trying to change `useImperialFormats`?

**Reply by paprophet** [reply]
> Show your code. Did you await `IModelApp.startup()` before trying to change `useImperialFormats`?


export class BasicViewportApp {   
 public static async startup() {  

  const quantityFormatter = new QuantityFormatter();
  quantityFormatter.useImperialFormats = false;   

  const opts: IModelAppOptions = {
     tileAdmin: TileAdmin.create({ useProjectExtents: false }),
     notifications: new ShowcaseNotificationManager(),
    toolAdmin: ShowcaseToolAdmin.initialize() ,
    quantityFormatter:quantityFormatter
  };

  await IModelApp.startup(opts);  

    const initPromises = new Array<Promise<any>>(); 
    initPromises.push(BasicViewportApp.initializeRpc());
    initPromises.push(IModelApp.i18n.registerNamespace("SimpleViewer").readFinished);
    initPromises.push(UiComponents.initialize(IModelApp.i18n));
    initPromises.push(Presentation.initialize({
      activeLocale: IModelApp.i18n.languageList()[0],
    }).then(() => {
      Presentation.selection.scopes.activeScope = "functional-element";
    
    }));
    await Promise.all(initPromises);  
  }

**Reply by pmconne** [reply]
Guess I should have looked at the implementation of `QuantityFormatter.useImperialFormats` - it assumes `IModelApp.startup()` has already been called.
Removing `quantityFormatter` from your `IModelAppOptions` and just setting `useImperialFormats` after awaiting `IModelApp.startup()` should do the trick.

**Reply by paprophet** [reply]
> Guess I should have looked at the implementation of `QuantityFormatter.useImperialFormats` - it assumes `IModelApp.startup()` has already been called.
> Removing `quantityFormatter` from your `IModelAppOptions` and just setting `useImperialFormats` after awaiting `IModelApp.startup()` should do the trick.

thank you
...
await IModelApp.startup(opts);  
IModelApp.quantityFormatter.useImperialFormats=false; 

success!

**Reply by ovaisa** [reply]
An update to the post:

'useImperialFormats' is now deprecated (as per v2.18.0 API documentation).  I am now able to set to Metric units by:

IModelApp.quantityFormatter.setActiveUnitSystem(false);
